rpc: TestRemoteOffsetUnhealthy failed

rpc.TestRemoteOffsetUnhealthy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11153765?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11153765?buildTab=artifacts#/) on master @ [8d2d2bf47cb3da1c27baa140a5afb5d28a5ed8f6](https://github.com/cockroachdb/cockroach/commits/8d2d2bf47cb3da1c27baa140a5afb5d28a5ed8f6):


```
=== RUN   TestRemoteOffsetUnhealthy
I230802 11:41:24.392821 435 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:36277,class=default,rpc] 19  connection is now healthy
I230802 11:41:24.399978 268 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:32905,class=default,rpc] 20  connection is now healthy
I230802 11:41:24.407282 426 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:35415,class=default,rpc] 21  connection is now healthy
I230802 11:41:24.413904 457 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:39329,class=default,rpc] 22  connection is now healthy
I230802 11:41:24.421564 472 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:32905,class=default,rpc] 23  connection is now healthy
I230802 11:41:24.428889 480 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:35415,class=default,rpc] 24  connection is now healthy
I230802 11:41:24.436021 500 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:39329,class=default,rpc] 25  connection is now healthy
I230802 11:41:24.442954 412 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:36277,class=default,rpc] 26  connection is now healthy
I230802 11:41:24.454246 512 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:35415,class=default,rpc] 27  connection is now healthy
I230802 11:41:24.460486 236 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:39329,class=default,rpc] 28  connection is now healthy
I230802 11:41:24.467946 237 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:36277,class=default,rpc] 29  connection is now healthy
I230802 11:41:24.475921 596 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:32905,class=default,rpc] 30  connection is now healthy
    context_test.go:1207: max offset: 100ms - node 0 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1207: max offset: 100ms - node 1 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1207: max offset: 100ms - node 2 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1201: max offset: 100ms - node 3 with excessive clock offset of 100.000001ms returned unexpected error: <nil>
--- FAIL: TestRemoteOffsetUnhealthy (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRemoteOffsetUnhealthy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

rpc.TestRemoteOffsetUnhealthy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11165192?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11165192?buildTab=artifacts#/) on master @ [602bc79cdfa99966093bb79755cf2b91b153475e](https://github.com/cockroachdb/cockroach/commits/602bc79cdfa99966093bb79755cf2b91b153475e):


```
=== RUN   TestRemoteOffsetUnhealthy
I230802 20:40:45.004963 320 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:35451,class=default,rpc] 22  connection is now healthy
I230802 20:40:45.014238 454 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:44811,class=default,rpc] 23  connection is now healthy
I230802 20:40:45.020052 562 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:37323,class=default,rpc] 24  connection is now healthy
I230802 20:40:45.025474 550 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:35451,class=default,rpc] 25  connection is now healthy
I230802 20:40:45.030879 560 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:34299,class=default,rpc] 26  connection is now healthy
I230802 20:40:45.036172 591 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:37323,class=default,rpc] 27  connection is now healthy
I230802 20:40:45.041613 626 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:35451,class=default,rpc] 28  connection is now healthy
I230802 20:40:45.047263 524 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:34299,class=default,rpc] 29  connection is now healthy
I230802 20:40:45.061812 659 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:44811,class=default,rpc] 30  connection is now healthy
    context_test.go:1207: max offset: 100ms - node 0 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1207: max offset: 100ms - node 1 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1207: max offset: 100ms - node 2 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1201: max offset: 100ms - node 3 with excessive clock offset of 100.000001ms returned unexpected error: <nil>
--- FAIL: TestRemoteOffsetUnhealthy (0.10s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRemoteOffsetUnhealthy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

rpc.TestRemoteOffsetUnhealthy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11173478?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11173478?buildTab=artifacts#/) on master @ [9cabe8c449e1a1b6ce0005cc1cf9567901dda1e2](https://github.com/cockroachdb/cockroach/commits/9cabe8c449e1a1b6ce0005cc1cf9567901dda1e2):


```
=== RUN   TestRemoteOffsetUnhealthy
I230803 11:42:08.345507 379 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:38483,class=default,rpc] 19  connection is now healthy
I230803 11:42:08.352183 400 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:43077,class=default,rpc] 20  connection is now healthy
I230803 11:42:08.358817 283 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:42331,class=default,rpc] 21  connection is now healthy
I230803 11:42:08.365106 439 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:43883,class=default,rpc] 22  connection is now healthy
I230803 11:42:08.371556 468 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:43077,class=default,rpc] 23  connection is now healthy
I230803 11:42:08.381158 452 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:42331,class=default,rpc] 24  connection is now healthy
I230803 11:42:08.387340 453 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:43883,class=default,rpc] 25  connection is now healthy
I230803 11:42:08.393862 300 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:38483,class=default,rpc] 26  connection is now healthy
I230803 11:42:08.400362 532 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:42331,class=default,rpc] 27  connection is now healthy
I230803 11:42:08.407404 217 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:43883,class=default,rpc] 28  connection is now healthy
I230803 11:42:08.414971 523 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:38483,class=default,rpc] 29  connection is now healthy
I230803 11:42:08.424268 585 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:43077,class=default,rpc] 30  connection is now healthy
    context_test.go:1207: max offset: 100ms - node 0 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1207: max offset: 100ms - node 1 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1207: max offset: 100ms - node 2 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1201: max offset: 100ms - node 3 with excessive clock offset of 100.000001ms returned unexpected error: <nil>
--- FAIL: TestRemoteOffsetUnhealthy (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRemoteOffsetUnhealthy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

rpc.TestRemoteOffsetUnhealthy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11173479?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11173479?buildTab=artifacts#/) on master @ [9cabe8c449e1a1b6ce0005cc1cf9567901dda1e2](https://github.com/cockroachdb/cockroach/commits/9cabe8c449e1a1b6ce0005cc1cf9567901dda1e2):


```
=== RUN   TestRemoteOffsetUnhealthy
I230803 11:43:26.257259 4877 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:38373,class=default,rpc] 19  connection is now healthy
I230803 11:43:26.263775 4969 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:40015,class=default,rpc] 20  connection is now healthy
I230803 11:43:26.272165 5017 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:46687,class=default,rpc] 21  connection is now healthy
I230803 11:43:26.278264 5109 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:46101,class=default,rpc] 22  connection is now healthy
I230803 11:43:26.283948 5187 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:40015,class=default,rpc] 23  connection is now healthy
I230803 11:43:26.290618 5300 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:46687,class=default,rpc] 24  connection is now healthy
I230803 11:43:26.296871 5402 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:46101,class=default,rpc] 25  connection is now healthy
I230803 11:43:26.303130 5542 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:38373,class=default,rpc] 26  connection is now healthy
I230803 11:43:26.309579 5657 2@rpc/peer.go:537  [rnode=4,raddr=127.0.0.1:46687,class=default,rpc] 27  connection is now healthy
I230803 11:43:26.319099 5817 2@rpc/peer.go:537  [rnode=1,raddr=127.0.0.1:46101,class=default,rpc] 28  connection is now healthy
I230803 11:43:26.325553 6032 2@rpc/peer.go:537  [rnode=2,raddr=127.0.0.1:38373,class=default,rpc] 29  connection is now healthy
I230803 11:43:26.331414 6175 2@rpc/peer.go:537  [rnode=3,raddr=127.0.0.1:40015,class=default,rpc] 30  connection is now healthy
    context_test.go:1207: max offset: 100ms - node 0 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1207: max offset: 100ms - node 1 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1207: max offset: 100ms - node 2 with acceptable clock offset of 0s did not return an error, as expected
    context_test.go:1201: max offset: 100ms - node 3 with excessive clock offset of 100.000001ms returned unexpected error: <nil>
--- FAIL: TestRemoteOffsetUnhealthy (0.10s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRemoteOffsetUnhealthy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/rpc/context.go
+++ b/pkg/rpc/context.go
@@ -213,9 +213,6 @@ type Context struct {
 	RemoteClocks *RemoteClockMonitor
 	MasterCtx    context.Context // cancel on stopper quiesce
 
-	heartbeatInterval time.Duration
-	heartbeatTimeout  time.Duration
-
 	rpcCompression bool
 
 	localInternalClient RestrictedInternalClient
@@ -531,13 +528,11 @@ func NewContext(ctx context.Context, opts ContextOptions) *Context {
 	secCtx.useNodeAuth = opts.UseNodeAuth
 
 	rpcCtx := &Context{
-		ContextOptions:    opts,
-		SecurityContext:   secCtx,
-		rpcCompression:    enableRPCCompression,
-		MasterCtx:         masterCtx,
-		metrics:           makeMetrics(),
-		heartbeatInterval: opts.RPCHeartbeatInterval,
-		heartbeatTimeout:  opts.RPCHeartbeatTimeout,
+		ContextOptions:  opts,
+		SecurityContext: secCtx,
+		rpcCompression:  enableRPCCompression,
+		MasterCtx:       masterCtx,
+		metrics:         makeMetrics(),
 	}
 
 	rpcCtx.dialbackMu.Lock()

--- a/pkg/rpc/context_test.go
+++ b/pkg/rpc/context_test.go
@@ -876,9 +876,6 @@ func TestOffsetMeasurement(t *testing.T) {
 	clientClock := &AdvancingClock{time: timeutil.Unix(0, 10)}
 	clientMaxOffset := time.Duration(0)
 	clientCtx := newTestContext(clusterID, clientClock, clientMaxOffset, stopper)
-	// Make the interval shorter to speed up the test.
-	clientCtx.RPCHeartbeatInterval = 1 * time.Millisecond
-	clientCtx.RPCHeartbeatTimeout = 1 * time.Millisecond
 	clientCtx.RemoteClocks.offsetTTL = 5 * clientClock.getAdvancementInterval()
 	if _, err := clientCtx.GRPCDialNode(remoteAddr, serverNodeID, DefaultClass).Connect(ctx); err != nil {
 		t.Fatal(err)
@@ -952,7 +949,7 @@ func TestFailedOffsetMeasurement(t *testing.T) {
 	clientCtx := newTestContext(clusterID, clock, maxOffset, stopper)
 	// Remove the timeout so that failure arises from exceeding the maximum
 	// clock reading delay, not the timeout.
-	clientCtx.heartbeatTimeout = 0
+	clientCtx.RPCHeartbeatTimeout = 0
 	// Allow two heartbeat for initialization. The first ping doesn't report an offset,
 	// the second one thus doesn't have an offset to work with, so it's only on the
 	// third one that's fully configured.
@@ -1022,9 +1019,6 @@ func TestLatencyInfoCleanupOnClosedConnection(t *testing.T) {
 	clientClock := &AdvancingClock{time: timeutil.Unix(0, 10)}
 	clientMaxOffset := time.Duration(0)
 	clientCtx := newTestContext(clusterID, clientClock, clientMaxOffset, stopper)
-	// Make the interval shorter to speed up the test.
-	clientCtx.RPCHeartbeatInterval = 1 * time.Millisecond
-	clientCtx.RPCHeartbeatTimeout = 1 * time.Millisecond
 
 	var hbDecommission atomic.Value
 	hbDecommission.Store(false)

--- a/pkg/rpc/datadriven_test.go
+++ b/pkg/rpc/datadriven_test.go
@@ -148,7 +148,6 @@ func setupEnv(t *testing.T) *ddEnv {
 	maxOffset := time.Duration(250)
 
 	clientCtx := newTestContext(clusterID, clock, maxOffset, stopper)
-	clientCtx.heartbeatInterval = 10 * time.Millisecond
 
 	env := &ddEnv{
 		clusterID: clusterID,

--- a/pkg/rpc/peer.go
+++ b/pkg/rpc/peer.go
@@ -184,8 +184,8 @@ func (rpcCtx *Context) newPeer(k peerKey) *peer {
 		dial: func(ctx context.Context, target string, class ConnectionClass) (*grpc.ClientConn, error) {
 			return rpcCtx.grpcDialRaw(ctx, target, class, rpcCtx.testingDialOpts...)
 		},
-		heartbeatInterval: rpcCtx.heartbeatInterval,
-		heartbeatTimeout:  rpcCtx.heartbeatTimeout,
+		heartbeatInterval: rpcCtx.RPCHeartbeatInterval,
+		heartbeatTimeout:  rpcCtx.RPCHeartbeatTimeout,
 	}
 	var b *circuit.Breaker
 

